src/lib.rs

```diff
@@ -358,7 +363,27 @@ impl fmt::Show for Error {
                 if other.is_empty() {
                     write!(f, "{}", usage)
                 } else {
-                    write!(f, "{}\n\n{}", other, usage)
+                    write!(f, "{:?}\n\n{}", other, usage)
+                }
+            }
+            Help => write!(f, ""),
+            NoMatch => write!(f, "Invalid arguments."),
+            Usage(ref s) | Argv(ref s) | Decode(ref s) | Version(ref s) => {
+                write!(f, "{}", s)
+            }
+        }
+    }
+}
+
+impl fmt::String for Error {
```

Comment:
Here, we could have removed `fmt::Show` entirely and relied only on `fmt::String`. However, `Result::unwrap` requires that the `Error` parameter implements `fmt::Show`, probably to generate an error message.